next/app/lib/components/Navbar.tsx

```diff
@@ -0,0 +1,109 @@
+"use client";
+
+import React from "react";
+import { Routes } from "../constants";
+import Link from "next/link";
+import { usePathname } from "next/navigation";
+import { User } from "@auth/core/types";
+import { FaAngleDown, FaAngleUp } from "react-icons/fa";
+import { keycloakSignOut } from "../actions/keycloak";
+import { Role } from "@/prisma/generated/client";
+
+export interface INavbarOption {
+  label: string;
+  route: string;
+  roles: Role[];
+}
+
+export interface INavbarProps {
+  user: User;
+}
+
+/** Client Component used for navigation */
+export const Navbar: React.FC<INavbarProps> = ({ user }) => {
+  const pathname = usePathname();
+  const [showUserDropDown, setShowUserDropDown] = React.useState(false);
+
+  const navItems: INavbarOption[] = [
+    { label: "Home", route: Routes.Home, roles: [Role.ZEVA_USER] },
+    {
+      label: "Compliance Reporting",
+      route: Routes.ComplianceReporting,
+      roles: [
+        Role.ZEVA_USER,
+        Role.DIRECTOR,
+        Role.SIGNING_AUTHORITY,
+        Role.ORGANIZATION_ADMINISTRATOR,
+      ],
+    },
+    {
+      label: "Credit Transactions",
+      route: Routes.CreditTransactions,
+      roles: [
+        Role.ZEVA_USER,
+        Role.DIRECTOR,
+        Role.SIGNING_AUTHORITY,
+        Role.ORGANIZATION_ADMINISTRATOR,
+      ],
+    },
```

Comment:
I think we need the ENGINEER_ANALYST role here as well, because a government user may have only that role and they would still need to be able to navigate to the credit transactions page.